### PR TITLE
CLN: remove deprecated DEF

### PR DESCRIPTION
**Pull request by fangchenli**
xref https://github.com/cython/cython/issues/4310

discovered in #47442

### CODE REVIEW

File: pandas/_libs/hashing.pyx [hunk 1]
```diff
@@ -21,8 +21,9 @@ import_array()
 
 from pandas._libs.util cimport is_nan
 
-DEF cROUNDS = 2
-DEF dROUNDS = 4
+cdef enum:
```
Comment (datapythonista) [inline pandas/_libs/hashing.pyx:24]: Seems like you require an extra blank line above this now.

File: pandas/_libs/hashing.pyx [hunk 2]
```diff
@@ -21,8 +21,10 @@ import_array()
 
 from pandas._libs.util cimport is_nan
 
-DEF cROUNDS = 2
-DEF dROUNDS = 4
+
+cdef enum:
```
Comment (mzeitlin11) [inline pandas/_libs/hashing.pyx:25]: Small nit: why is `enum` used here? - `cROUNDS` and `dROUNDS` just look like unrelated globals based on the usage below. While equivalent, maybe clearer to use an explicit `int` type?

**Review by mzeitlin11 — APPROVED** [review]

File: pandas/_libs/hashing.pyx [hunk 3]
```diff
@@ -162,6 +159,8 @@ cdef uint64_t low_level_siphash(uint8_t* data, size_t datalen,
     cdef uint8_t* end = data + datalen - (datalen % sizeof(uint64_t))
     cdef int left = datalen & 7
     cdef int left_byte
+    cdef int cROUNDS = 2
+    cdef int dROUNDS = 4
```
Comment (jbrockmendel) [inline pandas/_libs/hashing.pyx:163]: is there any upside to assigning this at the module level?

Comment (fangchenli) [reply pandas/_libs/hashing.pyx:163]: I don't see any in this case.

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @fangchenli
